### ISSUE TITLE
opt: optsteps panics with "could not find path"

### ISSUE THREAD

**Issue by RaduBerinde** [cockroachdb/cockroach#43091]
```
exec-ddl
CREATE TABLE customers (                        
    id INT8 NOT NULL,                           
    name STRING NOT NULL,                       
    address STRING NULL,                        
    CONSTRAINT "primary" PRIMARY KEY (id ASC),  
    FAMILY "primary" (id, name, address)        
)                                               
----


exec-ddl
CREATE TABLE orders (                                                                                       
    id INT8 NOT NULL,                                                                                       
    customer_id INT8 NULL,                                                                                  
    status STRING NOT NULL,                                                                                 
    CONSTRAINT "primary" PRIMARY KEY (id ASC),                                                              
    CONSTRAINT fk_customer_id_ref_customers FOREIGN KEY (customer_id) REFERENCES customers(id),             
    INDEX orders_auto_index_fk_customer_id_ref_customers (customer_id ASC),                                 
    FAMILY "primary" (id, customer_id, status),                                                             
    CONSTRAINT check_status CHECK (status IN ('open':::STRING, 'complete':::STRING, 'cancelled':::STRING))  
)                                                                                                           
----

optsteps
SELECT * FROM orders LEFT JOIN customers ON customer_id = customers.id
----
```

Panics with:
```
panic: could not find path to expr (in) [recovered]
	panic: could not find path to expr (in) [recovered]
	panic: could not find path to expr (in)

goroutine 27 [running]:
testing.tRunner.func1(0xc00022e500)
	/usr/local/go/src/testing/testing.go:830 +0x392
panic(0x16cd860, 0xc00036bc40)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/datadriven.runDirective.func1.1(0xc00053d978)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/datadriven/datadriven.go:267 +0x149
panic(0x16cd860, 0xc00036bc40)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/cockroachdb/cockroach/pkg/sql/opt/testutils/opttester.(*memoGroups).FindPath(0xc0001c27d0, 0x1cf3340, 0xc000562a98, 0x1cf2d80, 0xc000633860, 0x0, 0x1d061a0, 0xc0002972c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/testutils/opttester/memo_groups.go:90 +0x22d
github.com/cockroachdb/cockroach/pkg/sql/opt/testutils/opttester.(*forcingOptimizer).LookupPath(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/testutils/opttester/forcing_opt.go:117
github.com/cockroachdb/cockroach/pkg/sql/opt/testutils/opttester.(*optSteps).Next(0xc00073b340, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/testutils/opttester/opt_steps.go:135 +0x3a5
github.com/cockroachdb/cockroach/pkg/sql/opt/testutils/opttester.(*OptTester).OptSteps(0xc000514c00, 0x8, 0x18013b7, 0x8, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/testutils/opttester/opt_tester.go:889 +0xea
github.com/cockroachdb/cockroach/pkg/sql/opt/testutils/opttester.(*OptTester).RunCommand(0xc000514c00, 0x1d166e0, 0xc00022e500, 0xc00053d978, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/testutils/opttester/opt_tester.go:398 +0x1952
```

**Comment by RaduBerinde** [comment]
Ha, it turns out this is a normalization rule that runs when we build table check constraints, which aren't attached to any expression in the memo.

**Comment by RaduBerinde** [comment]
Fixed by #43405.
